tests/tt_metal/tt_metal/dispatch/dispatch_buffer/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+set(UNIT_TESTS_DISPATCH_BUFFER_SRC
+    ${CMAKE_CURRENT_SOURCE_DIR}/test_EnqueueWriteBuffer_and_EnqueueReadBuffer.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/test_sub_device.cpp
+    CACHE INTERNAL
```

Comment:
I initially made this a cache variable because I wanted it to be visible to the `CMakeLists.txt` file in the parent directory, and the `CACHE` keyword allows variables to be globally accessible, however, since it only needs to be visible to the parent, I've just changed it to use `PARENT_SCOPE` instead.

Reply:
Follow-up question: why does it need to be visible in the parent dir?  🙃 
In CMake generally "objects" should be encapsulated and know about themselves and connect to others like Lego pieces, rather than flinging variables around.

It looks like we're building 2 executables for each test?  One as a topic-focused, and then another "umbrella" executable that combines a bunch of topics?  Why do we want two binaries for the same test?

Currently that means each test file will be compiled twice.  Instead can create an `OBJECT` lib for the test files; they'll be compiled once.  Then define both executables listing the `OBJECT` targets as dependencies.  No duplicate compile, still two binaries, and bonus: no hucking variables to wider scopes.